bubblewrap.c

```diff
@@ -3155,23 +3173,53 @@ main (int    argc,
   if (opt_userns2_fd > 0 && setns (opt_userns2_fd, CLONE_NEWUSER) != 0)
     die_with_error ("Setting userns2 failed");
 
-  if (opt_unshare_user &&
-      (ns_uid != opt_sandbox_uid || ns_gid != opt_sandbox_gid) &&
-      opt_userns_block_fd == -1)
+  if (opt_unshare_user && opt_userns_block_fd == -1 &&
+      (ns_uid != opt_sandbox_uid || ns_gid != opt_sandbox_gid ||
+       opt_disable_userns))
```

Comment:
Note that as written here, we can get into this block even if we're setuid root. Please check that this isn't a security vulnerability?

I think it's OK, because we could already have entered this block anyway (for `--unshare-user --sandbox-uid=42`) even with a setuid bwrap, and the only thing we're doing here is reducing the `max_user_namespaces` in a new namespace, never increasing it.

Reply:
I don't see how this is ever a problem. We only ever hit this if --unshare-user is set, which means at this point we did CLONE_USERNS. So, even if we had privileged capabilities on the host we now don't have them in the user namespace:

>  On  the other hand, that process has no capabilities in the parent (in the case of [clone(2)](https://man7.org/linux/man-pages/man2/clone.2.html)) or previous (in the case of [unshare(2)] (https://man7.org/linux/man-pages/man2/unshare.2.html)  and [setns(2)](https://man7.org/linux/man-pages/man2/setns.2.html)) user namespace, even if the new namespace is  created or joined by the root user (i.e., a process with user ID 0 in the root namespace).

I.e. even in the setuid case we have at this point the same privileges as the non-setuid case.